Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Services/ApiGatewayRouteConfigService.cs

```diff
@@ -0,0 +1,221 @@
+            }
+
+            var jsonValue = entry.Value?.ToString()?.Trim();
+            _logger.LogDebug("Environment variable value: {VariableValue}", jsonValue);
+            if (string.IsNullOrEmpty(jsonValue))
+                continue;
+
+            try
+            {
+                if (jsonValue.StartsWith('['))
+                {
+                    _logger.LogDebug("Environment variable value starts with '['. Attempting to deserialize as a List.");
+                    var config = JsonSerializer.Deserialize<List<ApiGatewayRouteConfig>>(jsonValue);
+                    if (config != null)
+                    {
+                        _routeConfigs.AddRange(config);
+                        _logger.LogDebug("Environment variable deserialized and added to the existing configuration.");
+                    }
+                    else
+                    {
+                        _logger.LogDebug("Environment variable was not properly deserialized and will be skipped.");
+                    }
+                }
+                else
+                {
+                    _logger.LogDebug("Environment variable value does not start with '['.");
+                    var config = JsonSerializer.Deserialize<ApiGatewayRouteConfig>(jsonValue);
+                    if (config != null)
+                    {
+                        _routeConfigs.Add(config);
+                        _logger.LogDebug("Environment variable deserialized and added to the existing configuration.");
+                    }
+                    else
+                    {
+                        _logger.LogDebug("Environment variable was not properly deserialized and will be skipped.");
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error deserializing environment variable {key}: {ex.Message}");
+                _logger.LogDebug("Error deserializing environment variable {Key}: {Message}", key, ex.Message);
+            }
+        }
+    }
+
+    /// <summary>
+    /// API Gateway selects the route with the most-specific match, using the following priorities:
+    /// 1. Full match for a route and method.
+    /// 2. Match for a route and method with path variable.
+    /// 3. Match for a route and method with a greedy path variable ({proxy+}).
+    /// 
+    /// For example, this is the order for the following example routes:
+    /// 1. GET /pets/dog/1
+    /// 2. GET /pets/dog/{id}
+    /// 3. GET /pets/{proxy+}
+    /// 4. ANY /{proxy+}
+    /// 
+    /// For more info: https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-routes.html
+    /// </summary>
+    private void UpdateRouteConfigMetadataAndSorting()
+    {
+        _logger.LogDebug("Updating the metadata needed to properly sort the Lambda config");
+        foreach (var routeConfig in _routeConfigs)
+        {
+            if (routeConfig.Path.Contains("{proxy+}"))
```

Comment:
What if you have a resource path like `/resource/{id}/subsegment/{proxy+}` so you have both a proxy and a variable?

Reply:
This is now handled